apps/api/src/app/events/services/performance-service/index.ts

```diff
@@ -0,0 +1,139 @@
+    return process.env.NODE_ENV === 'test';
+  }
+
+  private publishMarks(showLogs: boolean): void {
+    if (showLogs) {
+      const eventsMarks = [
+        MarkFunctionNameEnum.CREATE_NOTIFICATION_JOBS,
+        MarkFunctionNameEnum.DIGEST_FILTER_STEPS,
+        MarkFunctionNameEnum.ENDPOINT_TRIGGER_EVENT,
+        MarkFunctionNameEnum.TRIGGER_EVENT,
+      ];
+
+      this.performanceService.marks.map((mark) => {
+        if (this.performanceService.filterMarks(mark, eventsMarks)) {
+          Logger.debug(mark, PerformanceContextEnum.CONTEXT_MARK);
+        }
+      });
+    }
+  }
+
+  private publishMeasures(showLogs: boolean): void {
+    const createNotificationJobsMeasures: number[] = [];
+    const digestFilterStepsMeasures: number[] = [];
+    const endpointTriggerEventMeasures: number[] = [];
+    const triggerEventMeasures: number[] = [];
+
+    this.performanceService.measures.map((measure) => {
+      if (measure.name.startsWith(MarkFunctionNameEnum.ENDPOINT_TRIGGER_EVENT)) {
+        endpointTriggerEventMeasures.push(measure.duration);
+      }
+
+      if (measure.name.startsWith(MarkFunctionNameEnum.CREATE_NOTIFICATION_JOBS)) {
+        createNotificationJobsMeasures.push(measure.duration);
+      }
+
+      if (measure.name.startsWith(MarkFunctionNameEnum.DIGEST_FILTER_STEPS)) {
+        digestFilterStepsMeasures.push(measure.duration);
+      }
+
+      if (measure.name.startsWith(MarkFunctionNameEnum.TRIGGER_EVENT)) {
+        triggerEventMeasures.push(measure.duration);
+      }
+
+      if (showLogs) {
+        Logger.debug(
+          `Duration: ${measure.duration.toFixed(2)} ms| Id: ${measure.name}`,
+          PerformanceContextEnum.CONTEXT_MEASURE
+        );
+      }
+    });
+
+    this.performanceService.calculateAverage(MarkFunctionNameEnum.ENDPOINT_TRIGGER_EVENT, endpointTriggerEventMeasures);
+    this.performanceService.calculateAverage(MarkFunctionNameEnum.TRIGGER_EVENT, triggerEventMeasures);
+    this.performanceService.calculateAverage(
+      MarkFunctionNameEnum.CREATE_NOTIFICATION_JOBS,
+      createNotificationJobsMeasures
+    );
+    this.performanceService.calculateAverage(MarkFunctionNameEnum.DIGEST_FILTER_STEPS, digestFilterStepsMeasures);
+  }
+
+  public setEnd(mark: IMark): void {
+    if (this.shouldExecute()) {
+      this.performanceService.setEnd(mark);
+    }
+  }
+
+  private setStart(mark: IMark): IMark {
+    if (this.shouldExecute()) {
+      this.performanceService.setStart(mark);
+    }
+
+    return mark;
+  }
+
+  public buildDigestFilterStepsMark(
+    transactionId: string,
+    templateId: string,
+    notificationId: string,
+    subscriberId: string
+  ): IMark {
+    const mark = {
+      id: `${MarkFunctionNameEnum.DIGEST_FILTER_STEPS}:event:${transactionId}:template:${templateId}:notification:${notificationId}:subscriber:${subscriberId}:steps`,
+    };
+
+    return this.setStart(mark);
+  }
+
+  public buildEndpointTriggerEventMark(transactionId: string): IMark {
```

Comment:
these all won't be needed if we will create generic function with mark like:
```
const mark = {
      id: `${functionName}:${uuid}`,
    };
```

Reply:
Addressing this comment here: https://github.com/novuhq/novu/pull/2878#discussion_r1117205847